lib/global-styles.php

```diff
@@ -162,6 +162,10 @@ function gutenberg_experimental_global_styles_get_stylesheet( $tree, $type = 'al
  * and enqueues the resulting stylesheet.
  */
 function gutenberg_experimental_global_styles_enqueue_assets() {
+	if ( ! gutenberg_experimental_global_styles_has_theme_json_support() ) {
```

Comment:
One of the advantages global styles brings is that themes don't need to create preset classes anymore. With this change I guess we remove this advantage right? I guess if theme.json is not present we don't need to make database queries and retrieve user settings but I guess we can still output preset classes?

Reply:
Yes, one of the goals of this PR is that we will no longer enqueue any CSS **if the theme doesn't have theme.json support**. This is a mechanism to make sure that 1) we don't interfere with the theme's CSS and 2) we don't load unnecessary styles. The current behavior has been reported as not ideal in core-editor by people. I still think it's valuable if the theme supports theme.json, though.